File: light-control/light_config.py
import json
import logging
import os.path

class LightConfig:
    def __init__(self, light_config_file_path='./light_config.json'):
        self.light_strips = []
        self.__logger = logging.getLogger()
        self.set_light_config_file_path(light_config_file_path)
        self.__load_light_config_file()

    def set_light_config_file_path(self, path):
        self.__light_config_file_path = path

    def num_strips(self):
        return len(self.light_strips)

    def get_light_strip(self, index):
        return self.light_strips[index]

    def __load_light_config_file(self):
        self.__logger.log(logging.DEBUG, 'Loading light config')
        if not os.path.exists(self.__light_config_file_path):
            raise Exception('Light Strip config required at light_config.json. See light_config_sample.json for an example')
        else:
            with open(self.__light_config_file_path, 'r') as json_config:
                data = json.load(json_config)
                self.light_strips = data['light_strips']

        self.__logger.log(logging.DEBUG, 'Light config loaded!')

File: light-control-ts/src/lightConfig.ts
import { readFileSync, existsSync } from 'fs'

export interface LightStripConfig {
  index_start: number
  index_end: number
  gpio_pin: number
  bpp: 3 | 4
  order: string
  skip: number[]
}

export class LightConfig {
  lightStrips: LightStripConfig[] = []

  constructor(lightConfigFilePath = './light_config.json') {
    if (!existsSync(lightConfigFilePath)) {
      throw new Error(
        'Light Strip config required at light_config.json. See light_config_sample.json for an example',
      )
    }
    const data = JSON.parse(readFileSync(lightConfigFilePath, 'utf-8')) as {
      light_strips: LightStripConfig[]
    }
    this.lightStrips = data.light_strips
  }
}
